### ISSUE TITLE
File includes don't work — and other issues

### ISSUE THREAD

**Issue by JaneSmith** [flathub/org.gnome.gitlab.johannesjh.favagtk#1]
Hello.

I've been using this app for quite a while, using the old manual Flatpak build method. I believe the package I was using was built on or before the 26th of August, 2021. That version worked fine.

Today, I uninstalled the manually-built package, and reinstalled via the new Flathub package. I'm now having several issues with the app.

**The first and most obvious issue, is that there seems to be a permissions issue with file includes.**

My Beancount file system involves multiple files. I have "index.beancount", which defines assets, expenses, commodities, the opening balance, etc. I have "price-history.beancount", which defines prices for different currencies or commodities. I also have files for each year, such as "2022.beancount", "2021.beancount", and so on. The "index.beancount" file is the master file, and includes all of the other files via include statements, such as:

`include "price-history.beancount"`
`include "2021.beancount"`
`include "2022.beancount"`

With the old manually-built package, I opened the "index.beancount" file, and everything worked as expected. I could even close the app and re-open it, and it would open the file automatically. This was all fantastic, and made for a very smooth and convenient experience.

With the new Flathub package, when I open the "index.beancount" file, it doesn't seem to open the included files. A lot of information is missing. There are errors listed, such as:

> File	Line	Error
> <load>	0	File glob "price-history.beancount" does not match any files
> <load>	0	File glob "2021.beancount" does not match any files
> <load>	0	File glob "2022.beancount" does not match any files

So, it's not working.

In an attempt to fix this, I opened Flatseal and added the `filesystem=host` permission, to give the app access to all files. This mostly fixed the problem. After manually opening "index.beancount", all the included files get loaded. However, when closing and re-opening the app, the "index.beancount" file gets loaded without the included files again. So, the automatic file-reopening is broken. I have to manually open the "index.beancount" file every time via the file open dialog.

**I have several other issues aside from this:**

- On the left sidebar, the "Errors" text vanishes when I mouse-over it or navigate to the Errors page, leaving only a red rectangle.
- On the Balance Sheet page, the "Line chart", "Area chart", "Treemap", and "Sunburst" buttons disappear entirely on mouse-over.
- After some browsing around, for example changing the display currency to "Converted to GBP", the entire app just freezes. No errors are displayed in the terminal though. I have to forcibly end the process.
- Finally, I note that when running the app from the terminal, a warning is displayed indicating that a development web server is being run instead of a production web server. I'm not sure how much of an issue this is.

However, I have no idea if these are issues with the FavaGTK Flathub package, FavaGTK in general, or Fava itself. Again, the version I was using previously was from August 2021 or older, so there may have been quite a lot of changes since then. That's why I'm lumping them all here... I'm not sure where to open separate issue reports for them. Please advise?

Thanks.

**Comment by johannesjh** [comment]
hi, thank you for reporting these issues (and sorry for the problems you are experiencing). 

note, I am moving the issue over to gnome's gitlab project since that is where I am doing the development.

* file includes do not work: yes, I noticed this as well. I will copy your report over to https://gitlab.gnome.org/johannesjh/favagtk/-/issues/3, and yes, I also think this has to do with flatpak filesystem sandboxing. 
* the other ui issues: I created a ticket over in the development repo https://gitlab.gnome.org/johannesjh/favagtk/-/issues/4
